### PR TITLE
Add menu background map

### PR DESCRIPTION
**Pull request by Jupeyy**
In fact, none of the 0.7 maps seem to work, i get buffer overflows all over.
For testing i just used Gold mine now.

since this is just #2731 without logic, and without any menu stuff, the best would be we just merge them somehow:

Animations working
Always forces design rendering(full detail etc.)
Regular maps work just fine

as said no logic yet, just display it for testing:

screenshot_2020-09-18_18-49-17

### CODE REVIEW

**Comment by Jupeyy** [comment]
we could also set the full logic into the new menu_background class, less ugly :D

**Comment by def-** [comment]
> since this is just #2731 without logic, and without any menu stuff, the best would be we just merge them somehow:

Do you want to try that or should I?

> In fact, none of the 0.7 maps seem to work, i get buffer overflows all over.

I guess we need a nice start menu map then.

**Comment by Jupeyy** [comment]
Sure, you can try if you want, I'm not hugely motivated, so it would take some time, if i do it :D

**Comment by Jupeyy** [comment]
finished the logic..

too improve it a bit compared to 0.7 i also added the possibility to add custom screen locations over for the mapper(the checkpoint numbers):

screenshot_2020-09-18_22-02-28


now the menus for changing the theme and the actual theme maps are only missing

**Comment by Jupeyy** [comment]
After loading the broken maps in 0.7 and saved them again, they worked.
they didnt use the cubic interpolation, it was probably just a bug.

We still need a autmn theme, bcs 0.7 doesnt provide one,
rest should work

**Comment by def-** [comment]
Looks really great.

jungle map looks a bit bland on DDNet tab, maybe we could choose a different location?

screenshot-20200919@083743

Let's take the autumn map by fokkonaut for now? https://github.com/teeworlds/teeworlds/pull/2682/files (I tried adding it but locations also look a bit wrong)

The themes location in menu looks fine to me in General, just add some spacing above it and make hte text for the themes smaller.

**Comment by def-** [comment]
@fokkonaut I noticed that for the autumn theme after a few minutes the clouds jump to a new location, not sure if that matters much though.

**Comment by def-** [comment]
I pushed 2 changes on top of this, I hope that's fine. If not, feel free to remove them.

**Comment by def-** [comment]
Main thing missing in my mind is the locations, they seem to be not so great on a few maps.

File: src/game/client/components/menus_settings.cpp [hunk 1]
```diff
@@ -1384,23 +1470,50 @@ void CMenus::RenderSettings(CUIRect MainView)
...
 		RenderSettingsGraphics(MainView);
+	}
 	else if(g_Config.m_UiSettingsPage == SETTINGS_SOUND)
+	{
+		m_pBackground->ChangePosition(CMenuBackground::POS_SETTINGS_SOUND);
 		RenderSettingsSound(MainView);
+	}
 	else if(g_Config.m_UiSettingsPage == SETTINGS_DDNET)
+	{
+		m_pBackground->ChangePosition(CMenuBackground::POS_SETTINGS_DDNET);
```
Comment (def-) [inline src/game/client/components/menus_settings.cpp:1514]: are these the same hard-codings as in vanilla? Because in 0.7 the locations look nicer.

**Comment by Jupeyy** [comment]
I only copied the location vanilla has, rest is pretty random. If purp tee has time today he'll look into a map.

I don't like the map from chillerdragon

**Comment by Jupeyy** [comment]
@def- they are now more inside the map all

Also for ppl who want new theme, they can control the whole locations with the tiles i wrote in an earlier message

**Comment by Jupeyy** [comment]
ah, the tiles in the maps are gone, bcs 0.6 cant load these tiles.. well then chillerdragons map should work.

btw @def- have you resaved the maps, bcs ^

**Comment by Jupeyy** [comment]
the logic shouldnt rely on Client()->RenderFrameTime()
when the client is inactive it will get huge and the camera is gone xd

**Comment by Jupeyy** [comment]
just clamp it

**Comment by def-** [comment]
Yes, I have resaved the autumn maps.

jungle still looks quite dark and monotone on DDNet tab I think.
screenshot-20200919@225519

**Comment by def-** [comment]
@ChillerDragon in autumn night the white spot in background in LAN tab blinks. Is that intentional?

**Comment by def-** [comment]
> ChillerDragon: and yea its intentional its a flickering lamp i thought it adds cool vibes

@Jupeyy Any idea why ChillerDragon can't respond here?

**Comment by Jupeyy** [comment]
probs blocked him bcs of mail spam xd, wait i unblock

**Comment by Jupeyy** [comment]
@def- idc, best is u just change the values, its always hard to get tastes right, i like the dark, if u want it more left just put a higher x value to
`m_Positions[POS_CUSTOM0 + i] = vec2(500.0f + (75.0f * (float)i), 1250.0f - (75.0f * (float)i));`
custom0 and 1 are ddnet and kog, u can also split them from the rest loop

**Comment by def-** [comment]
Then I guess it's ok as is. If others also don't like it, we can still change the positions later. Is this change ready to go?

**Comment by Jupeyy** [comment]
sure

**Comment by def-** [comment]
I'm obviously a big fan of this. Anyone else want to take a look before I merge it?

**Comment by edg-l** [comment]
This looks rly good.

File: src/base/system.c [hunk 2]
```diff
@@ -2263,6 +2263,46 @@ int time_timestamp(void)
...
+	case 6:
+	case 7:
+		return SEASON_SUMMER;
+	case 8:
+	case 9:
+	case 10:
+		return SEASON_AUTUMN;
+	}
+	return SEASON_SPRING; // should never happen
+}
```
Comment (heinrich5991) [inline src/base/system.c:2304]: This function doesn't work for Australia.

Not sure if we can do anything about that… I can't think of a better solution.

Comment (Jupeyy) [reply src/base/system.c:2304]: Maybe over timezone strings, that gives a rough location, but not standard and probs not reliable enough.
We could also split the languages into AUS english, US english etc. but probs not worth it.
Else probs by letting the user specify it only

Comment (def-) [reply src/base/system.c:2304]: Not important enough in my opinion. Some regions have no seasons, should we consider the seasons there? :D

Comment (Jupeyy) [reply src/base/system.c:2304]: btw not that we talk about seasons, should we renamed (automatic) to (seasons) or smth, automatic sounds more like random

Comment (def-) [reply src/base/system.c:2304]: Yeah, I guess (seasons) sounds good. Could also add a (random) btw, but just an idea.

**Comment by Jupeyy** [comment]
ok added random and changed automatic to season

**Comment by Jupeyy** [comment]
oh btw, can you pls remove that weird flickering in the autumn map :D

**Comment by def-** [comment]
Is that question directed at me or @ChillerDragon ?

**Comment by Jupeyy** [comment]
I don't know. Both probably

**Comment by def-** [comment]
Done.

**Comment by def-** [comment]
I plan to merge this later today if no one else wants to review.

**Comment by ChillerDragon** [comment]
@def- pending pr at https://github.com/Jupeyy/ddnet/pull/1

**Comment by def-** [comment]
bors r+
Biggest thanks for this, it looks just amazing.

**Pull request merged**